service/src/main/java/uk/gov/hmcts/reform/fpl/service/RepresentativeService.java

```diff
@@ -0,0 +1,174 @@
+                        otherPersonLabel, representativeLabel));
+                }
+            }
+
+            if (DIGITAL_SERVICE.equals(servingPreferences)) {
+                if (isEmpty(representative.getEmail())) {
+                    validationErrors.add(format("Enter an email address for %s", representativeLabel));
+                } else {
+                    if (organisationService.findUserByEmail(authorisation, representative.getEmail()).isEmpty()) {
+                        validationErrors.add(
+                            format("%s must already have an account with the digital service", representativeLabel));
+                    }
+                }
+            }
+        }
+
+        return validationErrors;
+    }
+
+    public void addRepresentatives(CaseData caseData, Long caseId, String auth) {
+        caseData.getRepresentatives()
+            .forEach(representative -> {
+                addToCase(representative, caseId, auth);
+                linkWithRepresentable(caseData, representative);
+            });
+    }
+
+    private void addToCase(Element<Representative> representativeWithId, Long caseId, String auth) {
+        Representative representative = representativeWithId.getValue();
+        if (DIGITAL_SERVICE.equals(representative.getServingPreferences()) && isNull(representative.getIdamId())) {
+            organisationService.findUserByEmail(auth, representative.getEmail()).ifPresent(
+                userId -> {
+                    caseService.addUser(auth, Long.toString(caseId), userId,
+                        representative.getRole().getCaseRoles());
+                    representative.setIdamId(userId);
+                }
+            );
+        }
+    }
+
+    private void linkWithRepresentable(CaseData caseData, Element<Representative> representative) {
+        findRepresentable(caseData, representative.getValue())
+            .ifPresent(representable -> representable.addRepresentative(representative.getId()));
+    }
+
+    private Optional<Representable> findRepresentable(CaseData caseData, Representative representative) {
+        switch (representative.getRole().getType()) {
+            case RESPONDENT:
+                return findRespondent(caseData, representative.getRole().getSequenceNo());
+            case OTHER:
+                return findOther(caseData, representative.getRole().getSequenceNo());
+            default:
+                return Optional.empty();
+        }
+    }
+
+    private Optional<Representable> findRespondent(CaseData caseData, int sequenceNo) {
+        List<Respondent> respondents = unwrapElements(caseData.getRespondents1());
+
+        return respondents.size() <= sequenceNo ? Optional.empty() : Optional.of(respondents.get(sequenceNo));
+    }
+
+    private Optional<Representable> findOther(CaseData caseData, int sequenceNo) {
+        List<Other> others = caseDataExtractionService.getOthers(caseData);
```

Comment:
As commented somewhere else this could be implemented in caseData maybe?

Reply:
done